x-pack/legacy/plugins/monitoring/server/plugin.js

```diff
@@ -0,0 +1,142 @@
+          expose: core.expose
+        }); // Route handlers depend on this for xpackInfo
+        await requireUIRoutes(core);
+      }
+    });
+
+    xpackMainPlugin.registerFeature({
+      id: 'monitoring',
+      name: i18n.translate('xpack.monitoring.featureRegistry.monitoringFeatureName', {
+        defaultMessage: 'Stack Monitoring',
+      }),
+      icon: 'monitoringApp',
+      navLinkId: 'monitoring',
+      app: ['monitoring', 'kibana'],
+      catalogue: ['monitoring'],
+      privileges: {},
+      reserved: {
+        privilege: {
+          savedObject: {
+            all: [],
+            read: []
+          },
+          ui: [],
+        },
+        description: i18n.translate('xpack.monitoring.feature.reserved.description', {
+          defaultMessage: 'To grant users access, you should also assign the monitoring_user role.'
+        })
+      }
+    });
+
+    const bulkUploader = initBulkUploader({
+      elasticsearchPlugin: plugins.elasticsearch,
+      xpackMainPlugin: plugins.xpack_main,
+      config,
+      log: core.log,
+      kbnServerStatus: kbnServer.status,
+      kbnServerVersion: kbnServer.version
+    });
+    const kibanaCollectionEnabled = config.get('xpack.monitoring.kibana.collection.enabled');
+    const { info: xpackMainInfo } = xpackMainPlugin;
+
+    if (kibanaCollectionEnabled) {
+      /*
+      * Bulk uploading of Kibana stats
+      */
+      xpackMainInfo.onLicenseInfoChange(() => {
+        // use updated xpack license info to start/stop bulk upload
+        const mainMonitoring = xpackMainInfo.feature('monitoring');
+        const monitoringBulkEnabled = mainMonitoring && mainMonitoring.isAvailable() && mainMonitoring.isEnabled();
+        if (monitoringBulkEnabled) {
+          bulkUploader.start(collectorSet);
+        } else {
+          bulkUploader.handleNotEnabled();
+        }
+      });
+    } else if (!kibanaCollectionEnabled) {
+      core.log(
+        ['info', LOGGING_TAG, KIBANA_MONITORING_LOGGING_TAG],
+        'Internal collection for Kibana monitoring is disabled per configuration.'
+      );
+    }
+
+    core.injectUiAppVars('monitoring', (core) => {
+      const config = core.config();
+      return {
+        maxBucketSize: config.get('xpack.monitoring.max_bucket_size'),
+        minIntervalSeconds: config.get('xpack.monitoring.min_interval_seconds'),
+        kbnIndex: config.get('kibana.index'),
+        showLicenseExpiration: config.get('xpack.monitoring.show_license_expiration'),
+        showCgroupMetricsElasticsearch: config.get('xpack.monitoring.ui.container.elasticsearch.enabled'),
+        showCgroupMetricsLogstash: config.get('xpack.monitoring.ui.container.logstash.enabled') // Note, not currently used, but see https://github.com/elastic/x-pack-kibana/issues/1559 part 2
+      };
+    });
+
+    // TODO
+    // This is not quite working but I don't know if we want to add a dependency
+    // on infra from the start....
```

Comment:
up to you. we have to add it in the new platform anyway, otherwise you won't have access to the plugin